tests/functional/pod_and_daemons/test_csi_logs_rotation.py

```diff
@@ -0,0 +1,178 @@
+            # test that new compressed file was added
+            return new_gz_logs_num == gz_logs_num + 1
+        else:
+            # the number of compressed file was already at maximum, so test that the old current log was compressed
+            # and the new one has smaller size
+            return new_current_log_file_size < current_log_file_size
+
+    def pump_logs_and_wait_for_rotation(self, pod_obj, logs_dir, log_file_name):
+        """
+        Tests that the log files on pod are rotated correctly
+
+        Args:
+            pod_obj (str): Pod object which is tested
+            logs_dir (str): Logs directory on this pod
+            log_file_name (str) Current log file name
+
+        """
+        log.info(f"Testing log {log_file_name} rotation on pod {pod_obj.name}")
+        gz_logs_num, current_log_file_size = pod_obj.get_logs_details_on_pod(
+            logs_dir, log_file_name
+        )
+        log.info(
+            f"Number of compressed logs = {gz_logs_num}, current log file size = {current_log_file_size}"
+        )
+
+        # pump current log file size
+        pod_obj.exec_cmd_on_pod(
+            command=f"-- truncate -s 560M {logs_dir + log_file_name}",
+            container_name="log-collector",
+            out_yaml_format=False,
+            shell=True,
+        )
+
+        time.sleep(10)  # wait fo make sure that the truncate had its effect
+        current_log_file_size = pod_obj.get_logs_details_on_pod(
+            logs_dir, log_file_name
+        )[1]
+        log.info(f"Current log file size after truncate is = {current_log_file_size}")
+
+        try:
+            for result in TimeoutSampler(
+                WAIT_FOR_ROTATION_TIME,
+                SLEEP_BETWEEN_TRIES,
+                self.logs_were_rotated,
+                pod_obj=pod_obj,
+                gz_logs_num=gz_logs_num,
+                current_log_file_size=current_log_file_size,
+                logs_dir=logs_dir,
+                log_file_name=log_file_name,
+            ):
+                if result:
+                    break
+            log.info("The logs were rotated correctly")
+        except TimeoutExpiredError:
+            assert False, "The logs were not rotated"
+
+    @pytest.mark.parametrize(
+        argnames=["pod_selector", "logs_dir", "log_file_name"],
+        argvalues=[
+            pytest.param(
+                *[
+                    "csi-cephfsplugin",
+                    "/var/lib/rook/openshift-storage.cephfs.csi.ceph.com/log/node-plugin/",
+                    "csi-cephfsplugin.log",
+                ],
+            ),
+            pytest.param(
+                *[
+                    "csi-rbdplugin",
+                    "/var/lib/rook/openshift-storage.rbd.csi.ceph.com/log/node-plugin/",
+                    "csi-rbdplugin.log",
+                ],
+            ),
+        ],
+    )
+    def test_pods_csi_log_rotation(self, pod_selector, logs_dir, log_file_name):
```

Comment:
What is the difference between 2 tests?, as per test name looks like both does log rotation for plugin pods.

Reply:
@ramkiperiy The difference between the two test cases is the type of the pods ( rbd plugin pod and cephfs plugin pod) , and therefore the path in the pod ( openshift-storage.rbd.csi.ceph.com vs openshift-storage.cephfs.csi.ceph.com) . This is two different test cases both of which need to be validated.

Reply:
@ramkiperiy  If I get you right, you ask why we need two tests with two test cases each and don't write one test with 4 test cases (2 test cases for plugin pod and 2 test cases for plugin provisioner)? I thought since the tests are very short ( a few lines) - such separation improves test readability. Would you still like me to merge it to be 1 test with 4 different test cases?

Reply:
@ramkiperiy Done.